substrate/primitives/core/src/ecdsa.rs

```diff
@@ -644,6 +644,19 @@ mod test {
 		assert_eq!(pair1.public(), pair2.public());
 	}
 
+	#[test]
+	fn generate_with_phrase_should_be_recoverable_with_from_string() {
+		let (pair, phrase, seed) = Pair::generate_with_phrase(None);
+		let repair_seed = Pair::from_seed_slice(seed.as_ref()).expect("seed slice is valid");
+		assert_eq!(pair.public(), repair_seed.public());
```

Comment:
```suggestion
		assert_eq!(pair.public(), repair_seed.public());
		assert_eq!(pair.secret, repair_seed.secret);
```

Reply:
same comment for this & subsequent two: why the hell would you check pubkey recovery, but not secret's recovery - false sense of fund security for the user? :P

Now, you could also check equality of the full `Pair`, but then you have to derive `PartialEq` and `Debug` on `Pair`.
I'm assuming it's a deliberate choice not to derive `Debug`, to avoid having it end up in a log accidentally, same reasoning as *not* implementing over-the-wire encoding for seeds?